miio/dreamevacuum_miot.py

```diff
@@ -0,0 +1,288 @@
+            "\n",
+            "Battery level: {result.battery_level}\n"
+            "Brush life level: {result.brush_life_level}\n"
+            "Brush left time: {result.brush_left_time}\n"
+            "Charging state: {result.charging_state.name}\n"
+            "Cleaning mode: {result.cleaning_mode.name}\n"
+            "Device fault: {result.device_fault.name}\n"
+            "Device status: {result.device_status.name}\n"
+            "Filter left level: {result.filter_left_time}\n"
+            "Filter life level: {result.filter_life_level}\n"
+            "Life brush main: {result.life_brush_main}\n"
+            "Life brush side: {result.life_brush_side}\n"
+            "Life sieve: {result.life_sieve}\n"
+            "Map view: {result.map_view}\n"
+            "Operating mode: {result.operating_mode.name}\n"
+            "Side cleaning brush left time: {result.brush_left_time2}\n"
+            "Side cleaning brush life level: {result.brush_life_level2}\n"
+            "Timer enabled: {result.timer_enable}\n"
+            "Timer start time: {result.start_time}\n"
+            "Timer stop time: {result.stop_time}\n"
+            "Voice package: {result.voice_package}\n"
+            "Volume: {result.volume}\n",
+        )
+    )
+    def status(self) -> DreameVacuumStatus:
+        """State of the vacuum."""
+
+        return DreameVacuumStatus(
+            {
+                prop["did"]: prop["value"] if prop["code"] == 0 else None
+                for prop in self.get_properties_for_mapping()
+            }
+        )
+
+    def send_action(self, siid, aiid, params=None):
+        """Send action to device."""
+
+        # {"did":"<mydeviceID>","siid":18,"aiid":1,"in":[{"piid":1,"value":2}]
+        if params is None:
+            params = []
+        payload = {
+            "did": f"call-{siid}-{aiid}",
+            "siid": siid,
+            "aiid": aiid,
+            "in": params,
+        }
+        return self.send("action", payload)
+
+    @command()
+    def start(self) -> None:
+        """Start cleaning."""
+        return self.send_action(3, 1)
+
+    @command()
+    def stop(self) -> None:
+        """Stop cleaning."""
+        return self.send_action(3, 2)
+
+    @command()
+    def home(self) -> None:
+        """Return to home."""
+        return self.send_action(2, 1)
+
+    @command()
+    def identify(self) -> None:
+        """Locate the device (i am here)."""
+        return self.send_action(17, 1)
+
+    @command()
+    def reset_mainbrush_life(self) -> None:
+        """Reset main brush life."""
+        return self.send_action(26, 1)
+
+    @command()
+    def reset_filter_life(self) -> None:
+        """Reset filter life."""
+        return self.send_action(27, 1)
+
+    @command()
+    def reset_sidebrush_life(self) -> None:
+        """Reset side brush life."""
+        return self.send_action(28, 1)
+
+    def get_properties_for_mapping(self) -> list:
+        """Retrieve raw properties based on mapping."""
```

Comment:
One minor change still to do. Could you please add here a note that this is a copy&paste from the base class to use different `max_properties` value? That needs to be cleaned up at some point to allow defining the wanted value as a class variable.

Reply:
done